docs/sources/operations/storage/retention.md

```diff
@@ -3,7 +3,157 @@ title: Retention
+  - Finally re-upload the new modified index files.
+
+Retention is applying directly on the index, however chunks are not deleted directly. They actually will be deleted by the compactor asynchronously in another loop (sweeper).
+
+Marked chunks will only  be deleted after `retention_delete_delay` configured is expired because:
+
+1. boltdb-shipper indexes are refreshed from the shared store on components using it (querier and ruler) at a specific interval. This means deleting chunks instantly could lead to components still having reference to old chunks and so they could fails to execute queries. Having a delay allows for components to refresh their store and so remove gracefully their reference of those chunks.
+
+2. It gives you a short period to cancel chunks deletion in case of mistakes.
+
+Marker files (containing chunks to deletes), should be store on a persistent disk, since this is the sole reference to them anymore.
+
+### Configuration
+
+The following compactor config shows how to activate retention with the compactor.
+
+```yaml
+compactor:
+  working_directory: /data/retention
+  shared_store: gcs
+  compaction_interval: 10m
+  retention_enabled: true
+  retention_delete_delay: 2h
+  retention_delete_worker_count: 150
+schema_config:
+    configs:
+      - from: "2020-07-31"
+        index:
+            period: 24h
+            prefix: loki_index_
+        object_store: gcs
+        schema: v11
+        store: boltdb-shipper
+storage_config:
+    boltdb_shipper:
+        active_index_directory: /data/index
+        cache_location: /data/boltdb-cache
+        shared_store: gcs
+    gcs:
+        bucket_name: loki
+```
+
+> Note that retention is only available if the index period is 24h.
+
+As you can see the retention needs `schema_config` and `storage_config` to access the storage.
+To activate retention you'll need to set `retention_enabled` to true otherwise the compactor will only compact tables.
+
+The `working_directory` is the directory where marked chunks and temporary tables will be saved.
+The `compaction_interval` dictates how often retention and/or compaction is applied, if it falls behind it will execute as soon as possible.
+`retention_delete_delay` is the delay after which the compactor will actually delete marked chunks.
+
+And finally `retention_delete_worker_count` setup the amount of goroutine to use to delete chunks.
+
+> Compaction and retention is idempotent, in case of restart the compactor will restart where he left off.
+
+#### Configuring retention period
+
+So far we've only talked about how to setup the compactor to apply retention, in this section we will explain how to configure the actual retention period to apply.
+
+Retention period is configure via the [`limits_config`](./../../../configuration/#limits_config) configuration section. It supports two type of retention `retention_stream` which is applied to all chunks matching the selector
```

Comment:
Change paragraph to

The retention period is configured within the [`limits_config`](./../../../configuration/#limits_config) configuration section. There are two ways of setting the `retention_stream`, which is applied to all chunks matching the selector.

Reply:
I realize something is missing here. Will fix.